homeassistant/components/vacuum/dyson.py

```diff
@@ -0,0 +1,217 @@
+"""
+Support for the Dyson 360 eye vacuum cleaner robot.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/vacuum.dyson/
+"""
+import asyncio
+import logging
+
+from homeassistant.components.vacuum import (SUPPORT_BATTERY,
+                                             SUPPORT_FAN_SPEED, SUPPORT_PAUSE,
+                                             SUPPORT_RETURN_HOME,
+                                             SUPPORT_STATUS, SUPPORT_STOP,
+                                             SUPPORT_TURN_OFF, SUPPORT_TURN_ON,
+                                             VacuumDevice)
+from homeassistant.util.icon import icon_for_battery_level
+from homeassistant.components.dyson import DYSON_DEVICES
+
+ATTR_FULL_CLEAN_TYPE = "full_clean_type"
+ATTR_CLEAN_ID = "clean_id"
+ATTR_POSITION = "position"
+
+DEPENDENCIES = ['dyson']
+
+_LOGGER = logging.getLogger(__name__)
+
+DYSON_360_EYE_DEVICES = "dyson_360_eye_devices"
+
+ICON = "mdi:roomba"
+
+SUPPORT_DYSON = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_PAUSE | \
+                SUPPORT_RETURN_HOME | SUPPORT_FAN_SPEED | SUPPORT_STATUS | \
+                SUPPORT_BATTERY | SUPPORT_STOP
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Dyson 360 Eye robot vacuum platform."""
+    _LOGGER.info("Creating new Dyson 360 Eye robot vacuum")
+    if DYSON_360_EYE_DEVICES not in hass.data:
+        hass.data[DYSON_360_EYE_DEVICES] = []
+
+    # Get Dyson Devices from parent component
+    from libpurecoollink.dyson_360_eye import Dyson360Eye
+    for device in [d for d in hass.data[DYSON_DEVICES] if
+                   isinstance(d, Dyson360Eye)]:
+        dyson_entity = Dyson360EyeDevice(hass, device)
+        hass.data[DYSON_360_EYE_DEVICES].append(dyson_entity)
+
+    add_devices(hass.data[DYSON_360_EYE_DEVICES])
+    return True
+
+
+class Dyson360EyeDevice(VacuumDevice):
+    """Dyson 360 Eye robot vacuum device."""
+
+    def __init__(self, hass, device):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.